

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="./">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Merge the topN/bottomN cell &mdash; STARK 1.0.1 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=03e43079" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=e59714d7" />

  
      <script src="_static/jquery.js?v=5d32c60e"></script>
      <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="_static/documentation_options.js?v=292eb321"></script>
      <script src="_static/doctools.js?v=9a2dae69"></script>
      <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
      <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
      <script>window.MathJax = {"tex": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true}, "options": {"ignoreHtmlClass": "tex2jax_ignore|mathjax_ignore|document", "processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
      <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Spatial Structure Capture Efficiency" href="ssce.html" />
    <link rel="prev" title="Accumlated Analysis" href="accum.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            STARK
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Overview</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="overview.html">What’s the STARK?</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">CLI_TUTORIAL</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="install.html">install the STARK</a></li>
<li class="toctree-l1"><a class="reference internal" href="count.html">preprocess the sc3DG</a></li>
<li class="toctree-l1"><a class="reference internal" href="model.html">3D model the sc3DG</a></li>
<li class="toctree-l1"><a class="reference internal" href="impute.html">Get imputed contact map</a></li>
<li class="toctree-l1"><a class="reference internal" href="nDS.html">Calculate the normlized detection score</a></li>
<li class="toctree-l1"><a class="reference internal" href="accum.html">Accumlated Analysis</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Merge the topN/bottomN cell</a></li>
<li class="toctree-l1"><a class="reference internal" href="ssce.html">Spatial Structure Capture Efficiency</a></li>
<li class="toctree-l1"><a class="reference internal" href="gini.html">calculate the GiniQC</a></li>
<li class="toctree-l1"><a class="reference internal" href="loop.html">detect the loop</a></li>
<li class="toctree-l1"><a class="reference internal" href="emptycells.html">EmptyCells</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">STARK</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Merge the topN/bottomN cell</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/merge.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="merge-the-topn-bottomn-cell">
<h1>Merge the topN/bottomN cell<a class="headerlink" href="#merge-the-topn-bottomn-cell" title="Link to this heading"></a></h1>
<style type="text/css">
body {
    text-align: justify;
}
</style><p>Given the sparsity of sc3DG-seq data, an integrated analysis of data from multiple
cells is often beneficial.
STARK facilitates this by enabling the aggregation of data from multiple single cells
for subsequent analysis. The aggregation is mathematically represented by the equation:</p>
<p><span class="math notranslate nohighlight">\(A\left(i,j\right)=\ \sum_{k=1}^{n}{a_k(i,j)}\)</span></p>
<p>Matrix A denotes the integrated 3D genome data from multiple cells, and a_k  represents the single-cell 3D genome data from the k-th cell. The indices i and j indicate the genomic positions of interactions. This aggregation function serves to generate a pseudo-bulk dataset that represents cell types or clusters of interest. The creation of pseudo-bulk data allows for the enhancement of signal-to-noise ratio and can reveal patterns that may not be evident when analyzing single cells in isolation. This approach is particularly useful for downstream analyses such as identifying consensus chromosomal structures, studying cell type–specific interactions, and exploring the organization of the genome across different cellular states.</p>
<p>If you want to integrate the sc3DG data in the entire directory, you can use the following command:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>stark<span class="w"> </span>--mcool<span class="w"> </span>./data/mcool<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--output<span class="w"> </span>./output/merge.mcool<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--resolution<span class="w"> </span><span class="m">10000</span>,400000,1000000<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--num<span class="w"> </span><span class="m">100</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--sort<span class="w"> </span>True<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--n_jobs<span class="w"> </span><span class="m">32</span>
</pre></div>
</div>
<p>else if you want to integrate the sc3DG data which you prefer, you can use the following command:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>stark<span class="w"> </span>--mcool<span class="w"> </span>./data/mcool<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--output<span class="w"> </span>./output/merge.mcool<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--resolution<span class="w"> </span><span class="m">10000</span>,400000,1000000<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--num<span class="w"> </span><span class="m">100</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--sort<span class="w"> </span>True<span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--n_jobs<span class="w"> </span><span class="m">32</span><span class="w"> </span><span class="se">\</span>
<span class="w">    </span>--select<span class="w"> </span>./data/select.txt
</pre></div>
</div>
<p>The select.txt file is a file that contains the names of the sc3DG data you want to integrate.
The file format is as follows:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>cell_1.mcool
cell_2.mcool
cell_3.mcool
</pre></div>
</div>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="accum.html" class="btn btn-neutral float-left" title="Accumlated Analysis" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="ssce.html" class="btn btn-neutral float-right" title="Spatial Structure Capture Efficiency" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024, Wulab.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>