

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="./">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Spatial Structure Capture Efficiency &mdash; STARK 1.0.1 documentation</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=03e43079" />
      <link rel="stylesheet" type="text/css" href="_static/css/theme.css?v=e59714d7" />

  
      <script src="_static/jquery.js?v=5d32c60e"></script>
      <script src="_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
      <script src="_static/documentation_options.js?v=292eb321"></script>
      <script src="_static/doctools.js?v=9a2dae69"></script>
      <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
      <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
      <script>window.MathJax = {"tex": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true}, "options": {"ignoreHtmlClass": "tex2jax_ignore|mathjax_ignore|document", "processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
      <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="calculate the GiniQC" href="gini.html" />
    <link rel="prev" title="Merge the topN/bottomN cell" href="merge.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            STARK
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Overview</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="overview.html">What’s the STARK?</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">CLI_TUTORIAL</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="install.html">install the STARK</a></li>
<li class="toctree-l1"><a class="reference internal" href="count.html">preprocess the sc3DG</a></li>
<li class="toctree-l1"><a class="reference internal" href="model.html">3D model the sc3DG</a></li>
<li class="toctree-l1"><a class="reference internal" href="impute.html">Get imputed contact map</a></li>
<li class="toctree-l1"><a class="reference internal" href="nDS.html">Calculate the normlized detection score</a></li>
<li class="toctree-l1"><a class="reference internal" href="accum.html">Accumlated Analysis</a></li>
<li class="toctree-l1"><a class="reference internal" href="merge.html">Merge the topN/bottomN cell</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Spatial Structure Capture Efficiency</a></li>
<li class="toctree-l1"><a class="reference internal" href="gini.html">calculate the GiniQC</a></li>
<li class="toctree-l1"><a class="reference internal" href="loop.html">detect the loop</a></li>
<li class="toctree-l1"><a class="reference internal" href="emptycells.html">EmptyCells</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">STARK</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Spatial Structure Capture Efficiency</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/ssce.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="spatial-structure-capture-efficiency">
<h1>Spatial Structure Capture Efficiency<a class="headerlink" href="#spatial-structure-capture-efficiency" title="Link to this heading"></a></h1>
<p>Spatial  Structure Capture Efficiency (SSCE) is introduced as a quantitative metric designed to evaluate the efficacy of single-cell sequencing in capturing the chromotin structural information. It is crucial to recognize that while the total number of sequenced contacts can vary significantly among cells, a lower count does not necessarily indicate inferior sequencing quality. Such a result may simply reflect the capture of a specific subset of the chromotin architecture, which can still have substantial biological implications. Conversely, a high number of detected contacts does not guarantee the biological relevance of these interactions or confirm an accurate representation of the cell’s chromotin spatial organization. With this in mind, we introduce a novel indicator, termed Spatial Structure Capture Efficiency (SSCE), which quantifies the proportion of intra-cellular contacts contributing to the formation of the cell’s chromotin architecture. This metric is essential for a more nuanced understanding of the structural fidelity of single-cell sequencing data.The SSCE are calculated as follows:
<span class="math notranslate nohighlight">\(SSCE=\frac{T+A}{I+C}\)</span>,
where  <span class="math notranslate nohighlight">\(T\)</span>, <span class="math notranslate nohighlight">\(A\)</span>, <span class="math notranslate nohighlight">\(I\)</span>, and <span class="math notranslate nohighlight">\(C\)</span> are derived from fitting the following regression:</p>
<p><span class="math notranslate nohighlight">\(C_i～B+I*inter_i+T*tad_i+A*ab_i\)</span></p>
<p>In this equation, <span class="math notranslate nohighlight">\(C_i\)</span> denotes the aggregate count of contacts associated with chromosome <span class="math notranslate nohighlight">\(i\)</span>, <span class="math notranslate nohighlight">\(inter_i\)</span> represents the number of interactions that chromosome <span class="math notranslate nohighlight">\(i\)</span> engages in with other chromosomes in the genome. <span class="math notranslate nohighlight">\(tad_i\)</span>  corresponds to the count of TAD boundaries identified on chromosome <span class="math notranslate nohighlight">\(i\)</span> at a resolution of 40 kilobases, which is pivotal for elucidating chromatin domain organization. <span class="math notranslate nohighlight">\(ab_i\)</span> signifies the count of A/B compartment switches for chromosome <span class="math notranslate nohighlight">\(i\)</span> determined at a resolution of 100 kilobases, providing insights into the chromatin’s compartmentalization. The constant <span class="math notranslate nohighlight">\(B\)</span> in the equation serves two critical roles. Firstly, it accounts for contacts that cannot be attributed to any specific chromosomal structure feature. Secondly, it includes contacts that may result from technical or systematic errors during the sequencing process. To ensure the accuracy of the model, any chromosomes where all variables are nullified are systematically excluded from the regression analysis.
Considering the inherent collinearity between the counts of <span class="math notranslate nohighlight">\(ab_i\)</span> and <span class="math notranslate nohighlight">\(tad_i\)</span>, we employ Ridge regression as our fitting method. The coefficients derived from the Ridge regression model are then utilized to compute the SSCE. It is logical to infer that higher values of structural features <span class="math notranslate nohighlight">\(ab_i\)</span> and <span class="math notranslate nohighlight">\(tad_i\)</span>, along with lower values of non-structural features <span class="math notranslate nohighlight">\(B\)</span> and <span class="math notranslate nohighlight">\(I\)</span>, contribute to higher SSCE scores. An elevated SSCE score, therefore, implies a more thorough and precise depiction of the single-cell chomotin structure.</p>
<p>To calculate the SSCE, you can use the following command:</p>
<div class="highlight-shell notranslate"><div class="highlight"><pre><span></span>stark<span class="w"> </span>ssce<span class="w"> </span>--mcool<span class="w"> </span>/absolute/path/to/data/test_ssce<span class="w"> </span><span class="se">\</span>
--genome<span class="w"> </span>/absolute/path/to/mm10/mm10.fa<span class="w"> </span><span class="se">\</span>
--output<span class="w"> </span>/absolute/path/to/result/ssce<span class="w"> </span><span class="se">\</span>
--nproc<span class="w"> </span><span class="m">2</span>
</pre></div>
</div>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="merge.html" class="btn btn-neutral float-left" title="Merge the topN/bottomN cell" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="gini.html" class="btn btn-neutral float-right" title="calculate the GiniQC" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024, Wulab.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>